Sources/PointFree/Admin/Index.swift

```diff
@@ -62,29 +62,18 @@ private let newEpisodeEmailRowView = View<Episode> { ep in
     ])
 }
 
-let sendNewEpisodeEmailMiddleware: Middleware<StatusLineOpen, ResponseEnded, T2<Episode.Id, Prelude.Unit>, Data> =
-  requireEpisode(notFoundMiddleware: redirect(to: .admin(.newEpisodeEmail(.show))))
-    <<< requireUser
+let sendNewEpisodeEmailMiddleware =
+  filterMap(require1 >>> pure, or: loginAndRedirect)
     <<< requireAdmin
+    <<< filterMap(fetchEpisode >>> pure, or: redirect(to: .admin(.newEpisodeEmail(.show))))
```

Comment:
I first made `filterMap` effectful to port over `requireEpisode` before realizing that it's a pure computation! Still going to be useful to apply `filterMap` to other kinds of effectful fetching, though.